### PR TITLE
Implement :tag commands in envelope buffer

### PR DESCRIPTION
**Pull request by lucc**
Proposed in #870.

### CODE REVIEW

**Comment by pazz** [comment]
Also: please squash  a6b185c and ee39bc7. Thanks!

**Comment by lucc** [comment]
Squashed.

Please have a look at the additional commit. I do copy the tags and it works. But I decided to not copy all tags. A hardcoded list of special tags is deliberatly excluded from the copy. Does that list make sense or are these just the tags that I use?

File: NEWS [hunk 1]
```diff
@@ -1,3 +1,6 @@
+next:
+* new command "tag" (and friends) in EnvelopeBuffer to add additional tags after sendout
```
Comment (dcbaker) [inline NEWS]: perhaps 'sending' instead of 'sendout'?

File: alot/commands/envelope.py [hunk 2]
```diff
@@ -141,7 +141,7 @@ def apply(self, ui):
             ui.notify(msg + ' to %s' % path)
             logging.debug('adding new mail to index')
             try:
-                ui.dbman.add_message(path, account.draft_tags)
+                ui.dbman.add_message(path, account.draft_tags+envelope.tags)
```
Comment (dcbaker) [inline alot/commands/envelope.py]: spaces around the `+` operator please

File: alot/commands/envelope.py [hunk 3]
```diff
@@ -570,3 +570,59 @@ def apply(self, ui):
...
+)
+class TagCommand(Command):
+
+    """manipulate message tags"""
+    repeatable = True
+
+    def __init__(self, tags=u'', action='add', **kwargs):
+        """
+        :param tags: comma separated list of tagstrings to set
+        :type tags: str
```
Comment (dcbaker) [inline alot/commands/envelope.py]: The type is listed here as a `str` but the default argument is `unicode`. Let's either set this to `basestring` or `unicode` and use an `assert` to ensure it is a unicode object.

**Comment by lucc** [comment]
applied comments and rebased

**Comment by dcbaker** [comment]
There's some test failures now. Either we need to fix the tests or mark that as expected failure. I'll leave it up to you decide what the right thing is.

**Comment by lucc** [comment]
Sorry, the fix was super easy (just add the `u` string prefix to the relevant test args).

**Comment by dcbaker** [comment]
no worries. I still havn't figured out what's wrong with travis on our branches. I've been disabling things until I get it to work, but I'm not quite sure what's wrong yet.

Either way, I think this is ready to land.

**Pull request merged**